From ddee9343d785a6c733532bc90d55bc6d6402eac3 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Mon, 13 Oct 2025 02:02:41 +0300
Subject: [PATCH 445/553] tst_QSettings: split the testEscapes() test:
 espcapedKeys

This is easier to debug.

Use example.org instead of software.org, the latter is regitered and we
shouldn't use it in tests.

Pick-to: 6.8 6.5
Change-Id: I1006bf1464aa7cc4d7bc4991f90202e416859200
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 65e0589350cfc82f8890427907ef7373990a7409)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../corelib/io/qsettings/tst_qsettings.cpp    | 46 ++++++++++++++-----
 1 file changed, 35 insertions(+), 11 deletions(-)

diff --git a/tests/auto/corelib/io/qsettings/tst_qsettings.cpp b/tests/auto/corelib/io/qsettings/tst_qsettings.cpp
index bbdfa297e9c..79782d25e9c 100644
--- a/tests/auto/corelib/io/qsettings/tst_qsettings.cpp
+++ b/tests/auto/corelib/io/qsettings/tst_qsettings.cpp
@@ -182,7 +182,10 @@ private slots:
     void childKeys();
     void testIniParsing_data();
     void testIniParsing();
+
     void testEscapes();
+    void testEscapedKeys_data();
+    void testEscapedKeys();
     void testNormalizedKey_data();
     void testNormalizedKey();
     void testVariantTypes_data() { populateWithFormats(); }
@@ -2829,10 +2832,6 @@ void tst_QSettings::testEscapes()
 {
     QSettings settings(QSettings::UserScope, "software.org", "KillerAPP");
 
-#define testEscapedKey(plainKey, escKey) \
-    QCOMPARE(iniEscapedKey(plainKey), QByteArray(escKey)); \
-    QCOMPARE(iniUnescapedKey(escKey), QString(plainKey));
-
 #define testUnescapedKey(escKey, plainKey, reescKey) \
     QCOMPARE(iniUnescapedKey(escKey), QString(plainKey)); \
     QCOMPARE(iniEscapedKey(plainKey), QByteArray(reescKey)); \
@@ -2873,13 +2872,6 @@ void tst_QSettings::testEscapes()
         QCOMPARE(v.toString(), QString(vStr)); \
     }
 
-    testEscapedKey("", "");
-    testEscapedKey(" ", "%20");
-    testEscapedKey(" 0123 abcd ", "%200123%20abcd%20");
-    testEscapedKey("~!@#$%^&*()_+.-/\\=", "%7E%21%40%23%24%25%5E%26%2A%28%29_%2B.-\\%5C%3D");
-    testEscapedKey(QString() + QChar(0xabcd) + QChar(0x1234) + QChar(0x0081), "%UABCD%U1234%81");
-    testEscapedKey(QString() + QChar(0xFE) + QChar(0xFF) + QChar(0x100) + QChar(0x101), "%FE%FF%U0100%U0101");
-
     testUnescapedKey("", "", "");
     testUnescapedKey("%20", " ", "%20");
     testUnescapedKey("/alpha/beta", "/alpha/beta", "\\alpha\\beta");
@@ -2947,6 +2939,38 @@ void tst_QSettings::testEscapes()
     testBadEscape("@Rect(1 2 3)", "@Rect(1 2 3)");
     testBadEscape("@@Rect(1 2 3)", "@Rect(1 2 3)");
 }
+
+void tst_QSettings::testEscapedKeys_data()
+{
+    QTest::addColumn<QString>("plainKey");
+    QTest::addColumn<QByteArray>("escKey");
+
+    QTest::newRow("empty-string") << u""_s << ""_ba;
+    QTest::newRow("space") << u" "_s << "%20"_ba;
+    QTest::newRow(" 0123 abcd ") << " 0123 abcd " << "%200123%20abcd%20"_ba;
+
+    QTest::newRow("special-characters")
+        << "~!@#$%^&*()_+.-/\\="
+        << "%7E%21%40%23%24%25%5E%26%2A%28%29_%2B.-\\%5C%3D"_ba;
+
+    const std::array arr1 = {QChar(0xabcd), QChar(0x1234), QChar(0x0081)};
+    QTest::newRow("qchar-array1") << QString(arr1) << "%UABCD%U1234%81"_ba;
+
+    const std::array arr2 = {QChar(0xFE), QChar(0xFF), QChar(0x100), QChar(0x101)};
+    QTest::newRow("qchar-array2") << QString(arr2) << "%FE%FF%U0100%U0101"_ba;
+}
+
+void tst_QSettings::testEscapedKeys()
+{
+    QFETCH(QString, plainKey);
+    QFETCH(QByteArray, escKey);
+
+    QSettings settings(QSettings::UserScope, "example.org", "KillerAPP");
+
+    QCOMPARE(iniEscapedKey(plainKey), escKey);
+    QCOMPARE(iniUnescapedKey(escKey), plainKey);
+}
+
 #endif
 
 void tst_QSettings::testCaseSensitivity()
-- 
2.51.2

